src/conn.c

```diff
@@ -1867,11 +1847,24 @@ static int _conn_connect(xmpp_conn_t *conn,
     if (!conn->domain)
         return XMPP_EMEM;
 
-    conn->sock = sock_connect(conn, host, port);
-    strophe_debug(conn->ctx, "xmpp", "sock_connect() to %s:%u returned %d",
-                  host, port, conn->sock);
-    if (conn->sock == -1)
-        return XMPP_EINT;
+    do {
+        conn->sock = sock_connect(conn->xsock);
+        if (conn->sock == -1)
```

Comment:
On Windows sock_t is unsigned:
```
../src/conn.c: In function '_conn_connect':
../src/conn.c:1852:24: warning: comparison of integer expressions of different signedness: 'sock_t' {aka 'unsigned int'} and 'int' [-Wsign-compare]
 1852 |         if (conn->sock == -1)
```

Reply:
This is something new to me since I don't have experience with windows. I guess I saw other places in the code which need to be fixed as well. Will fix it.

Reply:
done